Client/Frontend/Shields/ShareShieldsController.swift

```diff
@@ -0,0 +1,280 @@
+    
+    private let tab: Tab
+
+    private let contentStackView = UIStackView().then {
+        $0.axis = .vertical
+        $0.spacing = 3
+        $0.alignment = .fill
+        $0.layoutMargins = UX.defaultInset
+        $0.isLayoutMarginsRelativeArrangement = true
+    }
+    
+    var actionHandler: ((Action) -> Void)?
+    
+    // MARK: Lifecycle
+    
+    init(tab: Tab) {
+        self.tab = tab
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    @available(*, unavailable)
+    required init(coder: NSCoder) {
+        fatalError()
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = Strings.Share.shareScreenTitle
+    
+        doLayout()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        preferredContentSize = contentStackView.systemLayoutSizeFitting(
+            CGSize(width: view.bounds.size.width - UX.contentSizeChange,
+                   height: view.bounds.size.height - UX.contentSizeChange),
+            withHorizontalFittingPriority: .required,
+            verticalFittingPriority: .fittingSizeLevel
+        ).with {
+            $0.width += UX.contentSizeChange
+        }
+        
+        navigationController?.preferredContentSize = preferredContentSize
+    }
+        
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        
+        if #available(iOS 13.0, *) {
+            if traitCollection.userInterfaceStyle != previousTraitCollection?.userInterfaceStyle {
+                applyTheme(Theme.of(tab))
+            }
+        }
+    }
+    
+    // MARK: Internal
+    
+    private func doLayout() {
+        let shareTypeViewEmail = ShareTypeView(type: .email)
+        let shareEmailGesture = UITapGestureRecognizer(target: self, action: #selector(shareEmailClicked))
+        shareTypeViewEmail.addGestureRecognizer(shareEmailGesture)
+        
+        let shareTypeViewTwitter = ShareTypeView(type: .twitter)
+        let shareTwitterGesture = UITapGestureRecognizer(target: self, action: #selector(shareTwitterClicked))
+        shareTypeViewTwitter.addGestureRecognizer(shareTwitterGesture)
+        
+        let shareTypeViewFacebook = ShareTypeView(type: .facebook)
+        let shareFacebookGesture = UITapGestureRecognizer(target: self, action: #selector(shareFacebookClicked))
+        shareTypeViewFacebook.addGestureRecognizer(shareFacebookGesture)
+        
+        let shareTypeViewDefault = ShareTypeView(type: .default)
+        let shareDefaultGesture = UITapGestureRecognizer(target: self, action: #selector(shareDefaultClicked))
+        shareTypeViewDefault.addGestureRecognizer(shareDefaultGesture)
+         
+        view.addSubview(contentStackView)
+
+        contentStackView.snp.makeConstraints {
+            $0.top.leading.trailing.equalTo(view.safeAreaLayoutGuide)
```

Comment:
why no `bottom` constraint here?

Reply:
The reason is we use prefered size to determine controllers size here and If we use bottom constraint the segue animation will not look good cause it will be like bottom button is resizing itself